### PR TITLE
Add DOME 4.0 to the acknowledgement

### PR DESCRIPTION
**Pull request by superstar54**
This PR
- adds DOME 4.0 to the acknowledgement, both README and the docs page.
- update the arrangement of the logos.
- update the authors and app description.

### CODE REVIEW

**Comment by superstar54** [comment]
Hi @mikibonacci, @AndresOrtegaGuerrero, and @edan-bainglass, who has time to do a quick review on this PR?  Thanks!

File: setup.cfg [hunk 1]
```diff
@@ -66,7 +66,7 @@ aiida.workflows =
 
 [aiidalab]
 title = Quantum ESPRESSO
-description = Perform Quantum ESPRESSO calculations
+description = The Quantum ESPRESSO (QE) app is a web-based interface in AiiDAlab, enabling users to perform first-principles calculations and compute material properties (e.g., band structure) directly in their browser.
```
[the remark this answers is not in the document]
Comment (superstar54) [reply setup.cfg]: Thanks! I modified it as you suggested.

**Review by AndresOrtegaGuerrero — APPROVED** [review]
LGTM! thanks!

**Comment by superstar54** [comment]
@AndresOrtegaGuerrero Thanks for the quick review!

**Pull request merged**